### PR TITLE
add e2e test measuring operator metrics

### PR DESCRIPTION
**Pull request by juanvallejo**
Work in progress...
Wires an extended test for operators measuring their requests-per-second.

cc @deads2k @soltysh

### CODE REVIEW

**Comment by juanvallejo** [comment]
cc @smarterclayton

File: test/extended/operators/operators.go [hunk 1]
```diff
@@ -156,6 +182,56 @@ var _ = g.Describe("[Feature:Platform][Smoke] Managed cluster should", func() {
 			e2e.Failf("None of the required core operators are available")
 		}
 	})
+
+	g.It("should iterate through operator pods and detect higher-than-normal queries per second", func() {
```
Comment (smarterclayton) [inline test/extended/operators/operators.go]: You aren't a smoke test, you don't belong in here.  Move to a separate describe.

File: test/extended/operators/operators.go [hunk 2]
```diff
@@ -156,6 +182,56 @@ var _ = g.Describe("[Feature:Platform][Smoke] Managed cluster should", func() {
...
+		podURLGetter := &portForwardURLGetter{
+			Protocol:   "https",
+			Host:       "localhost",
+			RemotePort: "8443",
```
Comment (smarterclayton) [inline test/extended/operators/operators.go]: You have to use the exec client.

File: test/extended/operators/operators.go [hunk 3]
```diff
@@ -156,6 +182,56 @@ var _ = g.Describe("[Feature:Platform][Smoke] Managed cluster should", func() {
...
+			LocalPort:  "37587",
+		}
+
+		namespaces, err := oc.AdminKubeClient().CoreV1().Namespaces().List(metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		failures := []error{}
+		failedPods := []v1.Pod{}
+		for _, ns := range namespaces.Items {
```
Comment (smarterclayton) [inline test/extended/operators/operators.go]: Please, please, please, just write a prometheus query that gives you the average over time of QPS.  That would be a 2 line test.

**Comment by juanvallejo** [comment]
@smarterclayton thanks, I'm using the prometheus API to calculate per-operator QPS in the latest commit

File: test/extended/operators/operators.go [hunk 4]
```diff
@@ -158,6 +167,80 @@ var _ = g.Describe("[Feature:Platform][Smoke] Managed cluster should", func() {
 	})
 })
 
+var _ = g.Describe("[Feature:Platform] Managed cluster should", func() {
```
Comment (soltysh) [inline test/extended/operators/operators.go]: You have should here and in `g.It` unnecessarily.

File: test/extended/operators/operators.go [hunk 5]
```diff
@@ -158,6 +167,80 @@ var _ = g.Describe("[Feature:Platform][Smoke] Managed cluster should", func() {
...
+		// detect rate of requests for a given operator over a timespan of 4 mins
+		metricFormat := `sum(rate(apiserver_request_count{client="%s/v0.0.0 (linux/amd64) kubernetes/$Format"}[4m]))`
+
+		tests := map[string][]prometheus.MetricTest{}
+		for _, c := range operatorClientNames {
+			tests[fmt.Sprintf(metricFormat, c)] = []prometheus.MetricTest{{GreaterThanEqual: false, Value: maxQPSAllowed}}
+		}
+
+		prometheus.RunQueries(tests, oc, ns, execPodName, url, bearerToken)
+		e2e.Logf("Telemetry is enabled: %s", bearerToken)
```
Comment (soltysh) [inline test/extended/operators/operators.go]: Not needed.

File: test/extended/prometheus/prometheus_builds.go [hunk 6]
```diff
@@ -102,18 +102,18 @@ type prometheusResponseData struct {
 	Result     model.Vector `json:"result"`
 }
 
-type metricTest struct {
-	labels map[string]string
-	// we are not more precise (greater than only, or equal only) becauses the extended build tests
+type MetricTest struct {
```
Comment (soltysh) [inline test/extended/prometheus/prometheus_builds.go]: Those utils should just move to `test/extended/utils` and build tests should move to builds and use utils then. But that can be done in a followup PR.

**Pull request closed without merging**